local-cluster/src/integration_tests.rs

```diff
@@ -1,5 +1,21 @@
-#![allow(clippy::integer_arithmetic)]
+//! Code used by the integration tests in the `local-cluster/tests` folder.
+//!
+//! According to the cargo documentation, integration tests are compiled as individual crates.
+//!
+//!   https://doc.rust-lang.org/book/ch11-03-test-organization.html#the-tests-directory
+//!
+//! Putting shared code into the `tests/` folder, causes it to be recompiled for every test, rather
+//! than being compiled once when it is part of the main crate.  And, at the same time, unused code
+//! analysis is broken, as it is literally included.  If a shared function is not used in one of the
+//! integration tests, it will be flagged as unused, when compiling this test crate.
```

Comment:
Thank you for finding the Rust book section on this.
Seems like this suggestion is questionable, as it clearly does not work all that well in practice.
It is not uncommon to treat warnings as errors.
And unused code is actually a pretty useful warning.

For the `local-cluster` crate, there is already a `cluster_tests` module.
I wonder if this should have been a location for the code from `tests/common/mod.rs`...
But I still want to submit this PR, as it will help in the other repo.
We can then merge these two modules, if it turns out to be a reasonable thing to do.